### PR TITLE
RE-175 IPAM routes

### PR DESCRIPTION
**Pull request by saxmfone1**

### CODE REVIEW

File: ns1/__init__.py [hunk 1]
```diff
@@ -59,6 +59,33 @@ def records(self):
...
+        :rtype: :py:class:`ns1.rest.ipam.Vrfs`
+        """
+        import ns1.rest.ipam
+        return ns1.rest.ipam.Vrfs(self.config)
+
+    def scope_groups(self):
+        """
+        Return a new raw REST interface to vrf resources
+
+        :rtype: :py:class:`ns1.rest.ipam.Vrfs`
```
Comment (jwilner) [inline ns1/__init__.py]: doc needs update

File: ns1/ipam.py [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+        """
+        self._rest = Addresses(config)
+        self.config = config
+        self.prefix = prefix
+        self.type = type
+        self.vrf = vrf
+        self.id = id
+        self.children = []
+        self.report = {}
+        self.data = None
```
Comment (jwilner) [inline ns1/ipam.py]: This doesn't yet address the optional scope group association; it's captured in the API by `scope_group_id`, so maybe that means we should pass in an SG analogous to vrf?

File: ns1/ipam.py [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+            else:
+                return self
+
+        if self.id is None:
+            if self.prefix is None or self.type is None or self.vrf is None:
+                raise AddressException('Must at least specify an id or prefix, type and vrf')
+            else:
+                vrf_id = self.vrf.id
+                try:
+                    self.id = [address for address in self._rest.list() if address['prefix'] == self.prefix and address[
```
Comment (jwilner) [inline ns1/ipam.py]: This should be available as an API call soon if not yet already.

File: ns1/ipam.py [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+                  'time-servers', 'vendor-class-identifier'],
+        'dhcpv6': ['dns-servers']
+    }
+
+    def __init__(self, address_family, options):
+        """
+        Create the DHCP options class that can be used by the IPAM API
+        :param str address_family: This is either dhcpv4 or dhcpv6
+        :param dict options: This is a dict representing the options set. Valid options are listed in
+        :attr:`ns1.ipam.DHCPOptions.OPTIONS` and
```
Comment (jwilner) [inline ns1/ipam.py]: Doc cuts off.

**Review by jwilner — COMMENTED** [review]
lgtm, just some little things

File: ns1/ipam.py (hunk 2 above)
Comment (saxmfone1) [reply ns1/ipam.py]: Makes sense, just checked in a change to address this

File: ns1/ipam.py (hunk 3 above)
Comment (saxmfone1) [reply ns1/ipam.py]: You referring to search?

Comment (jwilner) [reply ns1/ipam.py]: Oh ya. Used to be called find by prefix.

**Review by ponbiki — APPROVED** [review]
LGTM (although I will be changing VRF routes in the near future per Mark)

**Review by mburtless — APPROVED** [review]
LGTM.  Not a blocker, but would be great to also get some test coverage for this in `tests/unit`

**Comment by saxmfone1** [comment]
> LGTM. Not a blocker, but would be great to also get some test coverage for this in `tests/unit`

Sounds good, basic units added

**Pull request merged**